src/main/java/com/example/car_dealership/repos/DealershipRepo.java

```diff
@@ -0,0 +1,87 @@
+package com.example.car_dealership.repos;
+
+import com.example.car_dealership.models.Vehicles;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+import javax.sql.DataSource;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+@Repository
+public class DealershipRepo {
+    @Autowired
+    private DataSource dataSource;
+
+    // Get all vehicles for a specific dealership
+    public List<Vehicles> getAllVehicles(int dealershipId) {
+        String query = "SELECT v.* FROM vehicles v " +
```

Comment:
I'm glad you decided to challenge yourself to let the user show vehicles based on dealership.